File: practices/interview-bit/bit-manipulation/different-bits-sum-pairwise.cpp
int Solution::cntBits(vector<int> &A) {
    long long ans = 0;
    int MOD = 1e9 + 7;
    for(int i = 0; i < 32; i++) {
        int cnt[2] = {0};
        for(int x : A) {
            cnt[(x & (1 << i)) != 0]++;
        }
        ans = (ans + (cnt[0] * 1LL * cnt[1]) % MOD) % MOD;
    }
    return (2 * ans) % MOD;
}

File: practices/interview-bit/bit-manipulation/different-bits-sum-pairwise.py
class Solution:
    def cntBits(self, A):
        n = len(A)
        ans = 0
        for i in range(0, 32):
            cnt0 = 0
            for j in range(0, n):
                if (A[j] & (1 << i)) == 0:
                    cnt0 += 1

            ans += 2 * cnt0 * (n - cnt0)

        return ans % (10**9 + 7)
